docs/content/direct/user-guide.md

```diff
@@ -0,0 +1,81 @@
+
+At its current level of development, KubeStellar has a simple but limited way to specify rule-based customization, called "template expansion".
+
+### Template Expansion
+
+Template expansion is an optional feature that a user can request on an object-by-object basis. The way to request this feature on an object is to put the following annotation on the object.
+
+```yaml
+    control.kubestellar.io/expand-templates: "true"
+```
+
+The customization that template expansion does when distributing an object from a WDS to a WEC is applied independently to each leaf string of the object and is based on the "text/template" standard pacakge of Go. The string is parsed as a template and then replaced with the result of expanding the template. Errors from this process are reported in the status field of the Binding object involved. Errors during template expansion usually produce broken YAML, in which case no corresponding object will be created in the WEC.
+
+The data used when expanding the template are properties of the WEC. These properties are collected from the following four sources, which are listed in decreasing order of precedence.
+
+1. The ConfigMap object, if any, that is in the namespace named "customization-properties" in the ITS and has the same name as the inventory object for the WEC. In particular, the ConfigMap string and binary data items whose name is valid as a [Go language identifier](https://go.dev/ref/spec#Identifiers) supply properties.
+1. The annotations of the inventory item for the WEC supply properties if the annotation's name (AKA key) is valid as a Go language identifier.
+1. The labels of the inventory item for the WEC supply properties if the label's name (AKA key) is valid as a Go language identifier.
+1. There is a pre-defined property whose name is "clusterName" and whose value is the name of the inventory item (i.e., the ManagedCluster object) for the WEC.
+
+A Binding object's `status` section has a field holding a slice of error message strings reporting user errors that arose the last time the transport controller processed that Binding, along with the `observedGeneration` reporting the `metadata.generation` that was processed. For each workload object that the Binding references: if template expansion reports errors for any destinations, the errors reported for the first such destination are included in the Binding object's status.
+
+Any failure in any template expansion for a given Binding suppresses propagation of desired state from that Binding; the previosly propagated desired state from that Binding, if any, remains in place in the WEC.
+
+Template expansion can only be applied when and where the un-expanded leaf strings pass the validation that the WDS applies, and can only epxress substring replacements.
```

Comment:
Not very clear. Also do you define what is a "leaf" before this place ?

Reply:
I rely on the general understanding of "leaf" as the things in a tree that have no subdivisions in the tree structure. I think readers can be relied on to understand that YAML and JSON documents have tree structure.